thread/std-compat.h

```diff
@@ -93,6 +91,17 @@ class thread {
         return m_th != nullptr;
     }
 
+    class id {
```

Comment:
```using id = std::tuple<photon::thread*>;```  may work?

Reply:
Does tuple support auto hash?

And I think a class is more clear.

Reply:
tuple has not hash, but it's clear by itself, as long as the reader know what is a tuple.